src/core/numerics/ion_updater/ion_updater.h

```diff
@@ -210,29 +207,28 @@ void IonUpdater<Ions, Electromag, GridLayout>::updateAndDepositAll_(Ions& ions,
     auto inDomainSelector
         = [&domainBox](auto const& part) { return core::isIn(cellAsPoint(part), domainBox); };
 
+    auto inGhostBox = [&ghostBox](auto& part) { return core::isIn(cellAsPoint(part), ghostBox); };
 
     // push domain particles, erase from array those leaving domain
     // push patch and level ghost particles that are in ghost area (==ghost box without domain)
     // copy patch and ghost particles out of ghost area that are in domain, in particle array
     // finally all particles in domain are to be interpolated on mesh.
 
-
     for (auto& pop : ions)
     {
         auto& domainParticles = pop.domainParticles();
 
-        auto domainPartRange = makeRange(std::begin(domainParticles), std::end(domainParticles));
+        auto domainPartRange = makeRange(domainParticles);
 
         auto firstOutside = pusher_->move(domainPartRange, domainPartRange, em, pop.mass(),
                                           interpolator_, inDomainSelector, layout);
 
         domainParticles.erase(firstOutside, std::end(domainParticles));
 
-
         auto pushAndCopyInDomain = [&](auto& particleArray) {
-            auto range = makeRange(std::begin(particleArray), std::end(particleArray));
-            auto firstOutGhostBox
-                = pusher_->move(range, range, em, pop.mass(), interpolator_, ghostSelector, layout);
+            auto range            = makeRange(particleArray);
+            auto firstOutGhostBox = pusher_->move(range, range, em, pop.mass(), interpolator_,
+                                                  inGhostBox, ghostSelector, layout);
```

Comment:
I think we should add a comment about why we give which selector in which position here. Why first inGhostBox and second ghostSelector (maybe also that one could now be renamed now because it's not super clear how it differs from the other at this line, for instance isInGhostArea)

Reply:
"inGhostLayer" ?

Reply:
sounds good